### PR TITLE
Add test for json object sorting as well as json array contains function

### PR DESCRIPTION
**Pull request by Jinxiao0302**
1. Added test cases for json object type sorting functions. The sorting results is displayed at `db/tests/records/operations/test_sort.py/test_json_sort_object`
2. Implemented the filtering functions of "json array contains" and "json array is not empty". Contain function is based on the sqlalchemy function ` jsonb_contains(jsonb, jsonb)`, which transforms input strings into jsonb type. The useage of either array ot string is to be discussed.

## Checklist
 
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/types/custom/json_array.py
Comment (dmos62) [inline db/types/custom/json_array.py:168]: I'd like the python class name to reflect that this function is about JSON, similarly to how the id string makes that clear.

Same goes for all function classes that are meant to operate only on JSON values.

**Review by dmos62 — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/stores/abstract-types/operations/filtering.ts
Comment (pavish) [inline mathesar_ui/src/stores/abstract-types/operations/filtering.ts:130]: For this filter, we do not expect any user input. So, this should be:
```suggestion
      [abstractTypeCategory.Array]: [],
```

**Review by pavish — CHANGES_REQUESTED** [review]
@Jinxiao0302 Looks good on the frontend part. I have one small suggested change.

**Review by pavish — APPROVED** [review]
Approving from the frontend side of changes.

@Jinxiao0302 Nice work, overall!

File: mathesar_ui/src/stores/abstract-types/operations/filtering.ts
Comment (Jinxiao0302) [reply mathesar_ui/src/stores/abstract-types/operations/filtering.ts]: Thanks for your suggestions, Pavish! I have made that change.

**Review by mathemancer — APPROVED** [review]
Okay, this is looking good. Thank you for your work on this, @Jinxiao0302 !

File: db/functions/base.py
Comment (dmos62) [inline db/functions/base.py:89]: Since `Noop` is so similar to `Literal`, could you explain in a docstring what it is necessary? I presume it's because `Literal` creates an SQLAlchemy expression like `literal(subexpression)`, which in some way doesn't play well with the subexpression being a list of literals? Thanks!

**Review by dmos62 — CHANGES_REQUESTED** [review]

File: db/functions/base.py
Comment (Jinxiao0302) [reply db/functions/base.py:89]: Thanks for your review! Since we are using array as a input in `JsonArrayContains`, it is having trouble passing the filtering tests because of the literal() function. Thus, we need another function without literal wrapping to have that porperly implemented.

Comment (mathemancer) [reply db/functions/base.py:89]: @Jinxiao0302 Given the confusion, I do think it makes sense to add a docstring indicating the reason for this DBFunction.
- The `Literal` DB function produces a `literal` SQLAlchemy wrapper which doesn't play nicely with the type conversion between python classes and db types in `psycopg2`. 

@dmos62 Long-term, we need to reassess how we're dealing with literal (in the general sense) values in the DBFunction subclasses. Wrapping an arg in `literal` should probably be avoided, except in situations where we actually want to produce a column of literals in the produced query. I don't think it should ever be used for the comparand in a filter.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Hey, @Jinxiao0302, I think all that remains before merging this is to add the requested docstring to the `Noop` class to avoid future confusion.

Thank you!

**Comment by Jinxiao0302** [comment]
Completed required changes.

**Comment by kgodey** [comment]
@Jinxiao0302 I disabled auto-merge because the PR won't merge until @mathemancer has approved it (since he requested changes earlier). I'm assigning this back to him to review.

**Review by mathemancer — APPROVED** [review]

**Comment by mathemancer** [comment]
@dmos62 I'm happy with this now. Feel free to merge after you re-review.

**Review by dmos62 — APPROVED** [review]

**Comment by dmos62** [comment]
Thanks for the contribution @Jinxiao0302!

**Pull request merged**
